### PR TITLE
History table

### PR DESCRIPTION
**Pull request by alekhrycaiko**
This is a PR for the main MVP of the history table, features included in this ticket are:
- history tabs based on date. Currently we have 3 supported tabs (today/within the last month and All).
- the history table component has 3 states that are supported. 
-- no data which displays a message
-- loading (whch has a loading spinner)
-- react table, which is a table of all given keys from the BE.
- Basic unit tests for the main FE components.
- I've setup the basic click handler that should determine what happens when a request is selected. No functionality should be expected at this time.

Things that need to be done prior to a merge:
- Testing on IE11 which I will do at home, I won't let this block my dev time in the meanwhile.
- Testing that within the last month works as expected. I haven't explicitly made data to test this but I will check prior to a merge.
- Need to talk with FE if we want to display Error messages from failed GET requests. I think we need to modify the Alert class a bit further if we want to do this. This is a low priority issue but it should get done sometime soon.

### CODE REVIEW

File: scripts/js/history_container.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    getRequestGenerator(date1, date2){
+        return "/requests?from=" + date1 + "&to=" +  date2;
+    }
+
+    componentWillMount(){
+        let dateObject = new Date();
+        let todayFormatted = moment().format("YYYY-MM-DD"); // can use this to construct range for "today".
+        let monthName = "Request's from " + moment().format("MMMM");
```
Comment (mandyfung) [inline scripts/js/history_container.jsx]: Small nit: "Request's" -> "Requests"

Comment (alekhrycaiko) [inline scripts/js/history_container.jsx]: Fixed!

**Comment by alekhrycaiko** [comment]
checked with webpack-dev-server on IE and it works, sadly dont have an easy way to get data on windows though x.x

**Comment by alekhrycaiko** [comment]
and yes moment.js is a library for date formatting and management, it's quite powerful and worth taking a look at!

File: scripts/js/history_table.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    accessor: "submittedBy"
+                },
+                {
+                    header: "lastModifiedBy",
+                    accessor: "lastModifiedBy"
+                },
+                {
+                    header : "dateModified",
+                    accessor: "dateModified"
+                }
```
Comment (mandyfung) [inline scripts/js/history_table.jsx]: Small nits: the headers can be changed to "Submitted By", "Last Modified By", "Last Modified Date"

Comment (alekhrycaiko) [inline scripts/js/history_table.jsx]: Sounds good, I've made the changes.

**Pull request merged**